### PR TITLE
pets runtime because she deserves it

### PR DESCRIPTION
**Pull request by ynot01**
# Document the changes in your pull request

bunch of runtimes i noticed during a live round, some spammier than others

added crash logs to some things to better understand why they are runtiming in the future

# Changelog

:cl:  
bugfix: fixed chem scan program looking weird until you scan something
/:cl:

### CODE REVIEW

File: code/datums/components/religious_tool.dm [hunk 1]
```diff
@@ -96,6 +96,9 @@
 			return
 		var/selection2type = easy_access_sect.rites_list[rite_select]
 		performing_rite = new selection2type(parent)
```
Comment (Bibby0110) [inline code/datums/components/religious_tool.dm:98]: won't it fail here?

Comment (ynot01) [reply code/datums/components/religious_tool.dm:98]: seemingly not
```
[2023-01-25 14:48:38.040] [01:03:38] runtime error: Cannot execute null.invoke effect().
 - proc name: AttemptActions (/datum/component/religious_tool/proc/AttemptActions)
 -   source file: religious_tool.dm,102
 -   usr: Poisons-The-Toast (/mob/living/carbon/human)
 -   src: /datum/component/religious_too... (/datum/component/religious_tool)
 -   usr.loc: the carpet (190,148,2) (/turf/open/floor/carpet/purple)
 -   call stack:
 - /datum/component/religious_too... (/datum/component/religious_tool): AttemptActions(the Altar of the Gods (/obj/structure/altar_of_gods), The Holy Book of Lord Singulot... (/obj/item/storage/book/bible/booze), Poisons-The-Toast (/mob/living/carbon/human), "icon-x=6;icon-y=4;left=1;butto...")
 - the Altar of the Gods (/obj/structure/altar_of_gods):  SendSignal("atom_attackby", /list (/list))
 - the Altar of the Gods (/obj/structure/altar_of_gods): attackby(The Holy Book of Lord Singulot... (/obj/item/storage/book/bible/booze), Poisons-The-Toast (/mob/living/carbon/human), "icon-x=6;icon-y=4;left=1;butto...")
 - the Altar of the Gods (/obj/structure/altar_of_gods): attackby(The Holy Book of Lord Singulot... (/obj/item/storage/book/bible/booze), Poisons-The-Toast (/mob/living/carbon/human), "icon-x=6;icon-y=4;left=1;butto...")
 - The Holy Book of Lord Singulot... (/obj/item/storage/book/bible/booze): melee attack chain(Poisons-The-Toast (/mob/living/carbon/human), the Altar of the Gods (/obj/structure/altar_of_gods), "icon-x=6;icon-y=4;left=1;butto...")
 - Poisons-The-Toast (/mob/living/carbon/human): ClickOn(the Altar of the Gods (/obj/structure/altar_of_gods), "icon-x=6;icon-y=4;left=1;butto...")
 - the Altar of the Gods (/obj/structure/altar_of_gods): Click(the floor (190,149,2) (/turf/open/floor/plasteel/dark), "mapwindow.map", "icon-x=6;icon-y=4;left=1;butto...")
```

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -197,7 +197,7 @@ Turf and target are separate in case you want to teleport some distance from a t
 	var/banned = C ? is_banned_from(C.ckey, "Appearance") : null
 
 	while(loop && safety < 5)
-		if(C && C.prefs.custom_names[role] && !safety && !banned)
+		if(C && C.prefs?.custom_names[role] && !safety && !banned)
```
Comment (ThatLing) [inline code/__HELPERS/unsorted.dm]: i dont want to fix prefs pr :(

Comment (ThatLing) [inline code/__HELPERS/unsorted.dm]: This is fixed in #17381

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: code/__HELPERS/unsorted.dm (hunk 2 above)
Comment (ynot01) [reply code/__HELPERS/unsorted.dm]: Sir yes sir oorah

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -198,6 +198,8 @@ Turf and target are separate in case you want to teleport some distance from a t
 
 	while(loop && safety < 5)
 		if(C && C.prefs.custom_names[role] && !safety && !banned)
+			if(!C.prefs)
```
Comment (ThatLing) [inline code/__HELPERS/unsorted.dm]: Remove these changes, the entire proc is reworked in tgui prefs

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: code/__HELPERS/unsorted.dm (hunk 3 above)
Comment (ynot01) [reply code/__HELPERS/unsorted.dm]: But my merge conflicts

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**